core/src/main/java/org/apache/iceberg/deletes/DVFileWriter.java

```diff
@@ -36,6 +36,22 @@ public interface DVFileWriter extends Closeable {
    */
   void delete(String path, long pos, PartitionSpec spec, StructLike partition);
 
+  /**
+   * Marks every position that is deleted in positionDeleteIndex as deleted in the given data file.
+   *
+   * @param path the data file path
+   * @param positionDeleteIndex the position delete index containing all the positions to delete
+   * @param spec the data file partition spec
+   * @param partition the data file partition
+   */
+  default void delete(
```

Comment:
We could also make this interface take in a List<Long> positions (kind of like a batch delete API on the writer) but it felt more natural to just pass in the index imo, cc @aokolnychyi @nastra thoughts?